scripts/release.sh

```diff
@@ -0,0 +1,278 @@
+            #if [ $record_version != $submodule_latest_version ]; then
+                echo
+                echo "Current version of $record_name: "$record_version
+                #echo "Latest version of $record_name: "$submodule_latest_version
+            #fi
+        fi
+    done <<< "$submodule_info"
+    echo
+    echo "Please update the submodules before continuing with the release."
+    echo "You must also edit the dependancy information in SmartDeviceLink-iOS.podspec"
+    read user_input 
+fi
+
+# 5 update changelog
+# TODO - insert a template into the changelog that includes the version the users have selected above.
+#echo "A template for this release has been inserted into the changelog.  Please update it."
+echo 
+echo "Please update CHANGELOG.md, then return here and press enter."
+read user_input
+
+
+# 6 generate documentation
+echo 
+prompt_user "Would you like to automatically generate documentation with Jazzy?"
+if [[ $? == 1 ]]; then
+    # 6 Install Jazzy
+    # Check if Jazzy is already installed, and if not then install jazzy
+    if [ -z "$(mdfind -name 'Jazzy')" ]; then
+        echo "Jazzy is not installed, attempting to install (may require your password)..."
+        sudo gem install jazzy
+    fi
+
+    # this runs Jazzy to generate the documentation
+    echo "Running Jazzy to generate documentation..."
+    jazzy --clean --objc --framework-root SmartDeviceLink --sdk iphonesimulator --umbrella-header SmartDeviceLink/public/SmartDeviceLink.h --theme theme --output docs
+fi
+
+# 7 Ensure that the RPC_SPEC has released to the master branch and update the submodule to point to the new release tag (or to the HEAD of master, if no release of the RPC_SPEC is occurring).
+echo 
+echo "Please check if there is a new release of the RPC_SPEC on https://www.github.com/smartdevicelink/rpc_spec"
+echo "If there is, please update the rpc_spec submodule to point to the newest commit on the master branch. Press enter to continue..."
+read user_input
+#TODO - phase ? - can this be automated.  Check version here.  Check version at site.
+#record_url=https://www.github.com/smartdevicelink/rpc_spec
+#submodule_latest_version=$(gh repo view $record_url --json latestRelease -q .latestRelease.tagName)
+
+echo
+echo "Please perform the following steps to set up the Github release."
+
+# git commands
+# 8 commit release to develop
+prompt_user "Would you like to commit these changes to the develop branch?"
+if [[ $? == 1 ]]; then
+    #todo - phase 3 - run the commands automatically
+    
+    # add changes to the commit
+    # -A is probably not correct.
+    git add -A
+    # create the commit
+    commit_message="Update for release $new_version_number"
+    git commit -m commit_message
+    # push the commit to develop
+    git push --set-upstream origin $develop_branch
+fi
+
+echo
+# 8 merge release to master
+prompt_user "Would you like to merge this release to master? (This will not push to master.)?"
```

Comment:
If the user says no to committing to develop, this should not run

Reply:
There is probably a lot more that should not run if you decide not to commit to develop.